lpcompiler.pas

```diff
@@ -1342,12 +1344,17 @@ function TLapeCompiler.ParseMethodHeader(out Name: lpString; addToScope: Boolean
         begin
           if addToScope then
             FStackInfo.addSelfVar(Lape_SelfParam, TLapeType(Typ));
+
+          methodDef := Result.MethodDef;
           Result.Free();
           Result := TLapeType_MethodOfType.Create(Self, TLapeType(Typ), nil, nil, '', @Pos);
+          Result.MethodDef := methodDef;
         end
         else if (not (Typ is TLapeType_SystemUnit)) then
           LapeException(lpeTypeExpected, [Tokenizer]);
-      end;
+      end
+      else if (Result.MethodDef = mdProperty) then
+        LapeException(lpeExpectedProperty, Tokenizer.DocPos);
```

Comment:
I don't see how that would make much sense, or have any uses. It would honestly just be confusing to have a regular none-typemethod being a property if you ask me, regular functions acting like variables, that would trow anyone off when looking for what variable it is/where it's declared.